cashctrl_api/cashed_client.py

```diff
@@ -64,120 +72,302 @@ def list_accounts(self) -> pd.DataFrame:
+        if result.empty and not allow_missing:
+            raise ValueError(f"No tax code found for id: {id}")
+        elif result.empty and allow_missing:
+            return None
         elif len(result) > 1:
-            raise ValueError(f"Multiple tax codes found for id {id}")
+            raise ValueError(f"Multiple tax codes found for id: {id}")
         else:
             return result.item()
 
-    def tax_code_to_id(self, name: str) -> int:
+    def tax_code_to_id(self, name: str, allow_missing = False) -> int | None:
         """
         Retrieve the id corresponding to a given tax code name.
 
         Args:
             name (str): The tax code name.
+            allow_missing (boolean): The flag to allow missing tax code.
 
         Returns:
-            int: The id associated with the provided tax code name.
+            int | None: The id associated with the provided tax code name.
+                        or None is allow_missing is True and there is no such tax code.
 
         Raises:
             ValueError: If no id is found for the given tax code name or if multiple ids are found.
         """
-        df = self.list_tax_codes()
+        df = self.list_tax_rates()
         result = df.loc[df['name'] == name, 'id']
-        if result.empty:
+        if result.empty and not allow_missing:
             raise ValueError(f"No id found for tax code {name}")
+        elif result.empty and allow_missing:
+            return None
         elif len(result) > 1:
             raise ValueError(f"Multiple ids found for tax code {name}")
         else:
             return result.item()
 
+
+    def invalidate_accounts_cache(self) -> None:
+        """
+        Invalidates the cached accounts data.
+        """
+        self._accounts_cache = None
+        self._accounts_cache_time = None
+
+    def invalidate_tax_rates_cache(self) -> None:
+        """
+        Invalidates the cached tax rates data.
+        """
+        self._tax_rates_cache = None
+        self._tax_rates_cache_time = None
+
+    def invalidate_currencies_cache(self) -> None:
+        """
+        Invalidates the cached currencies data.
+        """
+        self._currencies_cache = None
+        self._currencies_cache_time = None
+
+    def invalidate_account_categories_cache(self) -> None:
+        """
+        Invalidates the cached account categories data.
+        """
+        self._account_categories_cache = None
+        self._account_categories_cache_time = None
+
+    def invalidate_journal_entries_cache(self) -> None:
+        """
+        Invalidates the cached journal entries data.
+        """
+        self._journal_entries_cache = None
+        self._journal_entries_cache_time = None
+
+    def invalidate_files_cache(self) -> None:
+        """
+        Invalidates the cached files data.
+        """
+        self._files_cache = None
+        self._files_cache_time = None
+
     def _is_expired(self, cache_time: Optional[datetime]) -> bool:
```

Comment:
I personally prefer `datetime | None` over `Optional[datetime]`.
(Requires a newer python version, though, which is fine for us.)